src/Component/Symbolizer/LineEditor/LineEditor.tsx

```diff
@@ -194,93 +222,58 @@ export class LineEditor extends React.Component<LineEditorProps> {
-                color={color}
-                onChange={this.onColorChange}
-              />
-            </Form.Item>
-            <Form.Item
-              label={locale.widthLabel}
-              {...formItemLayout}
-            >
-              <WidthField
-                width={width}
-                onChange={this.onWidthChange}
-              />
-            </Form.Item>
-            <Form.Item
-              label={locale.opacityLabel}
-              {...formItemLayout}
-            >
-              <OpacityField
-                opacity={opacity}
-                onChange={this.onOpacityChange}
-              />
-            </Form.Item>
-            <Form.Item
-              label={locale.dashLabel}
-              {...formItemLayout}
-            >
-              <LineDashField
-                dashArray={dasharray}
-                onChange={this.onDasharrayChange}
-              />
-            </Form.Item>
-            <Form.Item
-              label={locale.dashOffsetLabel}
-              {...formItemLayout}
-            >
-              <OffsetField
-                offset={dashOffset}
-                onChange={this.onDashOffsetChange}
-                disabled={symbolizer.dasharray === undefined || _get(symbolizer, 'dasharray.length') === 0}
-              />
-            </Form.Item>
-            <Form.Item
-              label={locale.capLabel}
-              {...formItemLayout}
-            >
-              <LineCapField
-                cap={cap}
-                onChange={this.onCapChange}
-                />
-            </Form.Item>
-            <Form.Item
-              label={locale.joinLabel}
-              {...formItemLayout}
-            >
-              <LineJoinField
-                join={join}
-                onChange={this.onJoinChange}
-              />
-            </Form.Item>
-          </Panel>
-          <Panel header="Graphic Stroke" key="2">
-            <GraphicEditor
-              graphicTypeFieldLabel={locale.graphicStrokeTypeLabel}
-              graphic={graphicStroke}
-              graphicType={_get(graphicStroke, 'kind')}
-              onGraphicChange={this.onGraphicStrokeChange}
-            />
-          </Panel>
-          <Panel header="Graphic Fill" key="3">
-            <GraphicEditor
-              graphicTypeFieldLabel={locale.graphicFillTypeLabel}
-              graphic={graphicFill}
-              graphicType={_get(graphicFill, 'kind')}
-              onGraphicChange={this.onGraphicFillChange}
-            />
-          </Panel>
-        </Collapse>
-      </div>
-    );
+          <CompositionContext.Consumer>
+            {(composition: Compositions) => (
+                <div className="gs-line-symbolizer-editor" >
+                  <Collapse bordered={false} defaultActiveKey={['1']} onChange={(key: string) => (null)}>
+                    <Panel header="General" key="1">
+                      {this.handleComposition(composition, 'color', this.onColorChange, <ColorField />)}
```

Comment:
I think it woul be nice to extract these lines to `const` above the `return` to keep the render tree as clean as possible.

Reply:
Currently I do not think this is possible. The `CompositionContext.Consumer` component injects the context to its children. Right now, I do not see another way of accessing the context inside the render function. The old way of retrieving context (`this.context`) does not return the context, and the new way (`useContext(<Context>)` hook) does not work for classes.